mlflow/openai/autolog.py

```diff
@@ -323,23 +336,35 @@ async def _stream_output_logging_hook(stream: AsyncIterator) -> AsyncIterator:
             _logger.warning(f"Encountered unexpected error when ending trace: {e}", exc_info=True)
 
 
-def _process_last_chunk(span: LiveSpan, chunk: Any, inputs: dict[str, Any], output: list[Any]):
-    if _is_responses_final_event(chunk):
-        output = chunk.response
-    else:
-        # Reconstruct a completion object from streaming chunks
-        output = _reconstruct_completion_from_stream(output)
-
-        # Set usage information on span if available
-        if usage := getattr(chunk, "usage", None):
-            usage_dict = {
-                TokenUsageKey.INPUT_TOKENS: usage.prompt_tokens,
-                TokenUsageKey.OUTPUT_TOKENS: usage.completion_tokens,
-                TokenUsageKey.TOTAL_TOKENS: usage.total_tokens,
-            }
-            span.set_attribute(SpanAttributeKey.CHAT_USAGE, usage_dict)
-
-    _end_span_on_success(span, inputs, output)
+def _process_last_chunk(
+    span: LiveSpan,
+    chunk: Any,
+    inputs: dict[str, Any],
+    output: list[Any],
+    is_responses_api: bool,
+) -> None:
+    try:
+        if _is_responses_final_event(chunk):
+            output = chunk.response
+        elif is_responses_api:
+            output = _reconstruct_response_from_stream(output)
+        else:
```

Comment:
```suggestion
        elif _is_chat_completion(...):
```

Instead of `else`, can we check output is the chat completion?